### DISCUSSION TITLE
Is there a way to flash a key until the value is set?

### DISCUSSION THREAD

**Discussion by korruptcow** [nguyenquyhy/Flight-Tracker-StreamDeck#156] [General]
In flight sim. I want to be able to flash a key if a value is met. For example I have seen people with another app or plugin have the gear key flash while it's going up or down and then either the light is on or off after the flash. It would also be handy for Stall warnings on speed. Any ideas guys?

Thanks

**Comment by nguyenquyhy** [comment]
Stall and overspeed should be easy to do because there is a variable for that, so basically we just need a new checkbox to indicate that you want to flash between ON and OFF images. 
For gear, which variables are you suggesting us to use?
Also, are there any other cases you can think of and which variables will be used in those cases?

**Reply by korruptcow** [reply]
Hi there!

Love the plug in so far! Took me AGES to figure out how to use it because of the syntax and examples but we got there in the end.

Well, I saw A dude who wrote his own plugin for stream deck and when He pushed the gear button the key went flashing red back ground (or just flashing red light indicator) then when the gear was down the key went green. That is cool. I am using the GEAR TOTAL PCT EXTENDED==1 variable at the moment to have the "on image" showing. But I have toggled GEAR_TOGGLE. When you press it you hear the gear but it takes a few seconds for the ON button to light up. Would be great to get an in progress bit.

**_Other suggestions._**

1.

It would be cool to map two things on a key IE OVER SPEED WARNING and STALL WARNING so you can have one key that changes state or possibly an extremes set on a scale. So if a warning is shown it's on the screen otherwise a normal variable such as speed could be showing. But having at least to be able to show 2 variables on a "DATA" key. 

2.

Be able to increase the "DATA VALUE" display on the keys to a larger font. As the moment it's rather small and i display speed and alt on there and would like a bigger font for the values :)


I have other Questions about Keys. I wanted to know if we can do maths on the key for a variable. I wanted to get ETA to next waypoint but it's in seconds and the distance is in meters. I would like the TIME ETA and DISTANCE in KM or Miles. 

Great plugin by the way. I want to make a video once I feel it's all setup, nice. IT's almost there

**Reply by nguyenquyhy** [reply]
Yeah the plugin can be a bit confusing, so we try our best with a user guide https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/blob/master/docs/USERGUIDE.md and a discussion thread https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/discussions/128 to share profiles. Please feel free to share yours there.

> I am using the GEAR TOTAL PCT EXTENDED==1 variable at the moment to have the "on image" showing. But I have toggled GEAR_TOGGLE. When you press it you hear the gear but it takes a few seconds for the ON button to light up. Would be great to get an in progress bit.

Yes, my understanding is for that we will need a intermediate new state that relies on 2 intermediate variables. So I really want to know if this can be applied to anything other than Gear before I make the generic button looks even scarier.

> But having at least to be able to show 2 variables on a "DATA" key.

This will probably come once we support some kind of boolean operations (e.g. `IE OVER SPEED WARNING | STALL WARNING`). It's not too hard and won't need new textbox/checkbox, so I'll try to add that soon.

> I wanted to know if we can do maths on the key for a variable. I wanted to get ETA to next waypoint but it's in seconds and the distance is in meters. I would like the TIME ETA and DISTANCE in KM or Miles.

The plugin can't do maths right now, but if you just want different units, SimConnect can do unit conversion on its own; you just have to give it the unit you want. Right now there is no UI to provide unit, but it will come pretty soon.

**Reply by korruptcow** [reply]
> The plugin can't do maths right now, but if you just want different units, SimConnect can do unit conversion on its own; you just have to give it the unit you want. Right now there is no UI to provide unit, but it will come pretty soon.

Can you give an example on how to request that from a button? I want to have like a ETA: on the AP saying something like  ETA 12:22 mins. I think I will have to wait until string are supported I tried to display GPS NEXT ID which should be an ICAO but the feedback value was 0. But 90% of the things I tried have worked awesome so thank you.
